x-pack/plugins/security/server/deprecations/privilege_deprecations.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import type { SecurityLicense } from '../../common/licensing';
+import type {
+  PrivilegeDeprecationsRolesByFeatureIdRequest,
+  PrivilegeDeprecationsRolesByFeatureIdResponse,
+} from '../../common/model';
+import { transformElasticsearchRoleToRole } from '../authorization';
+import type { AuthorizationServiceSetupInternal, ElasticsearchRole } from '../authorization';
+import { getDetailedErrorMessage, getErrorStatusCode } from '../errors';
+
+export const getPrivilegeDeprecationsServices = (
+  authz: AuthorizationServiceSetupInternal,
```

Comment:
In this case we should probably be a bit more specific about the parameter type:
```suggestion
  authz: Pick<AuthorizationServiceSetupInternal, 'applicationName'>,
```